### PR TITLE
docs(bpp_static_obstacle_avoidance): add documentation

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

Add explanation for

- lateral margin params
- planning at red traffic signal

## Tests performed

Nothing.

## Effects on system behavior

Nothing.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md [hunk 1]
```diff
@@ -747,6 +747,59 @@ The longitudinal positions depends on envelope polygon, ego vehicle specificatio
...
+
+Basically, user had better make `hard_margin_for_parked_vehicle` larger than `hard_margin` to prevent collision with doors or people who suddenly get out from vehicle.
+
+On the other hand, this module has only one parameter `soft_margin` for soft lateral margin constraint.
+
+![fig](./images/path_generation/hard_margin.png)
+
+As the hard margin parameters the distance which the user definitely want to keep, they are used in the logic to check whether the ego can pass side of the target object without avoidance maneuver as well.
+
+If the lateral distance is less than `hard_margin`/`hard_margin_for_parked_vehicle` when assuming that the ego follows current lane without avoidance maneuver, this module thinks the ego can pass the side of the object safely and the ego must avoid it. In this case, this module inserts stop point until the avoidance maneuver is allowed to execute so that the ego can avoid the object after approval. (e.g. The ego keeps stopping in front of such a object until operator approves avoidance maneuver if user uses this module in MANUAL mode.)
```
Comment (go-sakayori) [inline planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md:783]: I couldn't understand the part "this module thinks the ego can pass the side of the object safely and the ego must avoid it". Is it safe when the lateral distance is less than hard_margin`/`hard_margin_for_parked_vehicle ? In my understanding this module thinks that the ego should pass the side of the object with a certain amount of lateral distance, and therefore  the ego must avoid the (parked) vehicle.

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md [hunk 2]
```diff
@@ -83,7 +83,7 @@ partition fillEgoStatus() {
 note right
   This module has following status:
   - RUNNING: target object is still remaining. Or, the ego hasn't returned original lane.
-  - CANCEL: taget obejct has gone. And, the ego hasn't initiated avoidance maneuver.
+  - CANCEL: taget object has gone. And, the ego hasn't initiated avoidance maneuver.
```
Comment (go-sakayori) [inline planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md:86]:
```suggestion
  - CANCEL: target object has gone. And, the ego hasn't initiated avoidance maneuver.
```

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md [hunk 3]
```diff
@@ -747,6 +747,59 @@ The longitudinal positions depends on envelope polygon, ego vehicle specificatio
...
+
+If the lateral distance is less than `hard_margin`/`hard_margin_for_parked_vehicle` when assuming that the ego follows current lane without avoidance maneuver, this module thinks the ego can pass the side of the object safely and the ego must avoid it. In this case, this module inserts stop point until the avoidance maneuver is allowed to execute so that the ego can avoid the object after approval. (e.g. The ego keeps stopping in front of such a object until operator approves avoidance maneuver if user uses this module in MANUAL mode.)
+
+![fig](./images/path_generation/must_avoid.png)
+
+On the other hand, if the lateral distance is larger than `hard_margin`/`hard_margin_for_parked_vehicle`, this module doesn't insert stop point even when it's waiting approval because it thinks it's possible to pass the side of the object safely.
+
+![fig](./images/path_generation/pass_through.png)
+
+### When there is no enough space
```
Comment (go-sakayori) [inline planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md:791]:
```suggestion
### When there is not enough space
```

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md [hunk 4]
```diff
@@ -747,6 +747,59 @@ The longitudinal positions depends on envelope polygon, ego vehicle specificatio
...
+
+This module inserts stop point only when the ego can potentially avoid the object. So, if it is not able to keep distance more than `hard_margin`/`hard_margin_for_parked_vehicle`, this module do nothing. Following figure shows the situation where this module is not able to keep enough lateral distance when user set a parameter `use_opposite_lane` to `false`.
```
Comment (go-sakayori) [inline planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md:793]:
```suggestion
This module inserts stop point only when the ego can potentially avoid the object. So, if it is not able to keep distance more than `hard_margin`/`hard_margin_for_parked_vehicle`, this module does nothing. Following figure shows the situation where this module is not able to keep enough lateral distance when user set a parameter `use_opposite_lane` to `false`.
```

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md [hunk 5]
```diff
@@ -747,6 +747,59 @@ The longitudinal positions depends on envelope polygon, ego vehicle specificatio
...
+
+![fig](./images/path_generation/do_nothing.png)
+
+!!! info
+
+    In this situation, obstacle stop feature in [obstacle_cruise_planner](../autoware_obstacle_cruise_planner/README.md) is responsible for ego vehicle safety.
```
Comment (go-sakayori) [inline planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md:799]: Link not found
```suggestion
    In this situation, obstacle stop feature in [obstacle_cruise_planner](../../autoware_obstacle_cruise_planner/README.md) is responsible for ego vehicle safety.
```

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md (hunk 2 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md]: @go-sakayori Thanks! I fixed it in https://github.com/autowarefoundation/autoware.universe/pull/7554/commits/dcdb6f5acf2ada1c030b31ffd16e92b56ca13c9c.

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md (hunk 3 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md]: https://github.com/autowarefoundation/autoware.universe/pull/7554/commits/dcdb6f5acf2ada1c030b31ffd16e92b56ca13c9c

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md (hunk 4 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md]: https://github.com/autowarefoundation/autoware.universe/pull/7554/commits/dcdb6f5acf2ada1c030b31ffd16e92b56ca13c9c

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md (hunk 5 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md]: https://github.com/autowarefoundation/autoware.universe/pull/7554/commits/dcdb6f5acf2ada1c030b31ffd16e92b56ca13c9c

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md (hunk 1 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md]: @go-sakayori Sorry, I wrote wrong explanation, so I fixed it in https://github.com/autowarefoundation/autoware.universe/pull/7554/commits/9938ae07674a4edff3d5fdbf688888a598c998bd.

WRONG: this module thinks the ego can pass the side of the object safely and the ego must avoid it.
CORRECT: this module thinks the ego can **NOT** pass the side of the object safely and the ego must avoid it.

**Review by go-sakayori — APPROVED** [review]

**Pull request merged**